packages/vite/src/node/server/index.ts

```diff
@@ -467,6 +467,13 @@ export async function _createServer(
         getDepsOptimizer(server.config, true)?.close(),
         closeHttpServer(),
       ])
+      while (server._pendingRequests.size > 0) {
+        await Promise.allSettled(
+          [...server._pendingRequests.values()].map(
+            (pending) => pending.request,
+          ),
+        )
+      }
```

Comment:
We also start awaiting for `pendingRequests`, and we throw early in `transformRequest` and in the hooks to restart faster, and to stop the import analysis plugin from keep calling transformRequest on all statically imported modules in the graph

Reply:
The `.map` is extracting the `request` field from `pending`, it is an object `{ timestamp, request }`. I'll add the comment